extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnChunkReaderImpl.java

```diff
@@ -47,19 +46,17 @@ public class ColumnChunkReaderImpl implements ColumnChunkReader {
 
     ColumnChunkReaderImpl(
             ColumnChunk columnChunk, SeekableChannelsProvider channelsProvider,
-            Path rootPath, ThreadLocal<CodecFactory> codecFactory, MessageType type,
+            Path rootPath, DeephavenCodecFactory codecFactory, MessageType type,
             OffsetIndex offsetIndex, List<Type> fieldTypes) {
         this.channelsProvider = channelsProvider;
         this.columnChunk = columnChunk;
         this.rootPath = rootPath;
         this.path = type
                 .getColumnDescription(columnChunk.meta_data.getPath_in_schema().toArray(new String[0]));
         if (columnChunk.getMeta_data().isSetCodec()) {
-            decompressor = ThreadLocal.withInitial(() -> codecFactory.get().getDecompressor(
-                    CompressionCodecName.valueOf(columnChunk.getMeta_data().getCodec().name())));
+            decompressor = codecFactory.getByName(columnChunk.getMeta_data().getCodec().name());
```

Comment:
Shouldn't we be blowing up here if we get a `null` result?

Reply:
Sure, I'll put a check in getByName, to make it throw if it doesnt find a match?